apps/minifront/src/state/swap/helpers.ts

```diff
@@ -74,73 +75,71 @@ export const sendCandlestickDataRequests = async (
-          correctedCandle.close = 1 / inverseCandle.close;
-          // high and low swap places
-          correctedCandle.high = 1 / inverseCandle.low;
-          correctedCandle.low = 1 / inverseCandle.high;
-          return correctedCandle;
-        },
-      ),
-    ],
-    ({ height }) => height,
-  );
-
-  // combine data at each height into a single candle
-  const combinedCandles = Array.from(collatedByHeight.entries()).map(
-    ([height, candlesAtHeight]) => {
+  return dexClient.candlestickData({ pair: { start, end }, limit, startHeight });
+};
+
+export const combinedCandlestickDataSelector = (
+  zQueryState: AbridgedZQueryState<{
+    direct: CandlestickDataResponse;
+    inverse: CandlestickDataResponse;
+  }>,
+) => {
+  if (!zQueryState.data) {
+    return { ...zQueryState, data: undefined };
+  } else {
+    const direct = zQueryState.data.direct.data;
+    const corrected = zQueryState.data.inverse.data.map(
+      // flip inverse data to match orientation of direct data
+      inverseCandle => {
+        const correctedCandle = inverseCandle.clone();
+        // comparative values are reciprocal
+        correctedCandle.open = 1 / inverseCandle.open;
+        correctedCandle.close = 1 / inverseCandle.close;
+        // high and low swap places
+        correctedCandle.high = 1 / inverseCandle.low;
+        correctedCandle.low = 1 / inverseCandle.high;
+        return correctedCandle;
+      },
+    );
+
+    // combine data at each height into a single candle
+    const combinedCandles = Array.from(
+      // collect candles at each height
+      Map.groupBy([...direct, ...corrected], ({ height }) => height),
+    ).map(([height, candlesAtHeight]) => {
       // TODO: open/close don't diverge much, and when they do it seems to be due
       // to inadequate number precision. it might be better to just pick one, but
       // it's not clear which one is 'correct'
-      const combinedCandleAtHeight = candlesAtHeight.reduce((acc, cur) => {
-        // sum volumes
-        acc.directVolume += cur.directVolume;
-        acc.swapVolume += cur.swapVolume;
-
-        // highest high, lowest low
-        acc.high = Math.max(acc.high, cur.high);
-        acc.low = Math.min(acc.low, cur.low);
-
-        // these accumulate to be averaged
-        acc.open += cur.open;
-        acc.close += cur.close;
-        return acc;
-      }, new CandlestickData({ height }));
+      const combinedCandleAtHeight = candlesAtHeight.reduce(
+        (acc, cur) => {
+          // sum volumes
+          acc.directVolume += cur.directVolume;
+          acc.swapVolume += cur.swapVolume;
+
+          // highest high, lowest low
+          acc.high = Math.max(acc.high, cur.high);
+          acc.low = Math.min(acc.low, cur.low);
+
+          // these accumulate to be averaged
+          acc.open += cur.open;
+          acc.close += cur.close;
+          return acc;
+        },
+        new CandlestickData({ height, low: Infinity, high: -Infinity }),
```

Comment:
This is probably my own ignorance, but shouldn't `low` and `high` be swapped? i.e., 
```suggestion
        new CandlestickData({ height, low: -Infinity, high: Infinity }),
```

Reply:
no, this is the 'initial' value of the reduce, which are intended to be clobbered by the first arriving values.

`Math.min(Infinity, x)` always returns x, `Math.max(-Infinity, y)` always returns y. but `Math.min(undefined, x)` returns `NaN`.